src/pages/index.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from "react";
+import { ethers } from "ethers";
+
+import {
+  CONTRACT_FACTORY_ADDRESS,
+  CONTRACT_FACTORY_ABI,
+  ROKY_ABI,
+  TOKEN_ADDRESS,
+  TOKEN_ABI,
+  ADMIN_ADDRESS
+} from "./contractData"; 
+
+export default function Home() {
+  const [provider, setProvider] = useState(null);
+  const [signer, setSigner] = useState(null);
+  const [userAddress, setUserAddress] = useState(null);
+  const [rokyAddress, setRokyAddress] = useState(null);
+  const [transferRecipient, setTransferRecipient] = useState(""); 
+
+  const FEE = 10; 
+  const SALT = "0x" + "0".repeat(62) + "01";
+
+  async function connectWallet() {
+    if (!window.ethereum) {
+      alert("MetaMask не найден");
+      return;
+    }
+    const browserProvider = new ethers.BrowserProvider(window.ethereum);
+    await browserProvider.send("eth_requestAccounts", []);
+    const signer = await browserProvider.getSigner();
+    const address = await signer.getAddress();
+    setProvider(browserProvider);
+    setSigner(signer);
+    setUserAddress(address);
+    alert("Кошелек подключен: " + address);
+  }
+
+  async function getOrCreateRoky() {
+    const factory = new ethers.Contract(CONTRACT_FACTORY_ADDRESS, CONTRACT_FACTORY_ABI, signer);
+    const predictedAddress = await factory.getRokyAddress(
+      userAddress,
+      TOKEN_ADDRESS,
+      ADMIN_ADDRESS,
+      FEE,
+      SALT
+    );
+
+    const code = await provider.getCode(predictedAddress);
+    if (code !== "0x") {
+      setRokyAddress(predictedAddress);
+      alert("Контракт уже существует по адресу: " + predictedAddress);
+      return predictedAddress;
+    }
+    const tx = await factory.createRoky(userAddress, TOKEN_ADDRESS, FEE, SALT);
+    const receipt = await tx.wait();
+
+    const event = receipt.logs.find(log => log.fragment?.name === "RokyDeployed");
+    const createdAddress = event?.args?.roky;
+    setRokyAddress(createdAddress || predictedAddress);
```

Comment:
эти адреса будут одинаковые, поэтому можешь сразу ставить createdAddress
```suggestion
    setRokyAddress(createdAddress);
```